django/db/models/fields/related_lookups.py

```diff
@@ -83,6 +83,29 @@ def as_sql(self, compiler, connection):
         else:
             return super(RelatedIn, self).as_sql(compiler, connection)
 
+    def __getstate__(self):
+        from django.db.models.query import QuerySet
+        obj_dict = self.__dict__.copy()
+        del obj_dict['rhs']
+        if isinstance(self.rhs, QuerySet):
+            # Pickle query fron queryset only
+            return (self.rhs.model, self.rhs.query, obj_dict)
+        else:
+            # Pickle RHS as is
+            return (None, self.rhs, obj_dict)
+
+    def __setstate__(self, state):
+        model, rhs, obj_dict = state
+        if model is None:
+            # Unpickle full RHS
+            self.rhs = rhs
+        else:
+            # Unpickle RHS as query
+            queryset = model.objects.none()
+            queryset.query = rhs
+            self.rhs = queryset
```

Comment:
Is this branching necessary? I can see how using `model.objects.none()` as a query holder could be problematic since it's not necessarily the same `QuerySet` class as the one from which `query` was extracted. Does the following work:

```python
def __getstate__(self):
    state = self.__dict__.copy()
    if isinstance(self.rhs, QuerySet):
        state['rhs'] = self.rhs.query
    return state
```